srcpkgs/Aegisub/template

```diff
@@ -1,34 +1,39 @@
 # Template file for 'Aegisub'
 pkgname=Aegisub
-version=3.3.2
-revision=11
-build_style=cmake
-build_helper=cmake-wxWidgets-gtk3
-cmake_builddir="BUILD"
-configure_args="-DwxWidgets_CONFIG_EXECUTABLE=$WX_CONFIG
- -DWITH_PORTAUDIO=$(vopt_if portaudio ON OFF)
- -DWITH_OPENAL=$(vopt_if openal ON OFF) -DWITH_FFTW3=$(vopt_if fftw ON OFF)"
-hostmakedepends="gettext-devel intltool pkg-config git LuaJIT"
-makedepends="alsa-lib-devel boost-devel libcurl-devel fontconfig-devel
- freetype-devel hunspell-devel icu-devel libass-devel libffms2-devel
- MesaLib-devel wxWidgets-gtk3-devel LuaJIT-devel $(vopt_if fftw fftw-devel)
- $(vopt_if openal libopenal-devel) $(vopt_if portaudio portaudio-devel)"
-short_desc="Subtitle editor (ssa, ass, srt)"
-maintainer="Orphaned <orphan@voidlinux.org>"
-license="BSD-3-Clause, MIT"
-homepage="https://github.com/wangqr/Aegisub"
-distfiles="https://github.com/wangqr/Aegisub/archive/v${version}.tar.gz"
-checksum="9aaab8e66cef2a72368c6b7f79b2f99700321a7b9a1ca623c5e6be0e65418db5"
+version=3.4.2
+revision=1
+build_style=meson
+configure_args="-Dsystem_luajit=true"
+hostmakedepends="pkg-config git gettext meson ninja LuaJIT wxWidgets-gtk3-devel"
+makedepends="libass-devel zlib-devel icu-devel boost-devel
+ wxWidgets-gtk3-devel MesaLib-devel LuaJIT-devel fontconfig-devel ffmpeg6
+ ffmpeg6-devel libffms2-devel fftw-devel hunspell-devel uchardet-devel gtest-devel
+ pulseaudio-devel alsa-lib-devel libopenal-devel portaudio-devel libcurl-devel"
+short_desc="Cross-platform advanced subtitle editor"
+maintainer="Komeil Parseh <komeilparseh@disroot.org>"
+license="BSD-3-Clause"
+homepage="https://github.com/TypesettingTools/Aegisub"
+distfiles="${homepage}/archive/v${version}.tar.gz"
+checksum="38731ec76392fe763ce34e34cda6ee233802cf20ef3f380a3998da35a893a5c4"
 
-build_options="fftw openal portaudio"
-build_options_default="fftw"
+CXXFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
 
-desc_option_fftw="Enable support for rendering of audio waveforms/spectrum"
-desc_option_openal="Enable support for openal"
+if [ "$cross_build" ]; then
+	configure_args+=" -Dwx_config=/usr/bin/wx-config-gtk3"
+fi
 
-CXXFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
+if [[ $XBPS_TARGET_MACHINE == arm* ]]; then
+	broken="Broken on ARM: wxWidgets C++ (operator[] ambiguity) and musl-devel issues."
```

Comment:
Hitting a persistent operator[] overloading error in wx/string.h during armv7l cross-compilation of Aegisub 3.4.2. Tried common CXXFLAGS workarounds (e.g., `DwxOVERLOAD_INT_ARRAYS=0`), but the error persists. This seems to be a known issue with wxWidgets/GCC on ARM. Any guidance or known solutions would be appreciated!

Reply:
Just to clarify, are you suggesting that `wxHAS_RESTRICTED_OVERLOADS` is being misconfigured during cross-build?

Reply:
No its missing the `wxSIZE_T_IS_UINT` but its seems like there is no generated config header that would define it and instead there is some `ifdef` heuristic to determine it based on the platform.